backend/src/test/java/ai/verta/modeldb/GlobalSharingTest.java

```diff
@@ -0,0 +1,451 @@
+    datasetServiceStub = DatasetServiceGrpc.newBlockingStub(channel);
+    client2DatasetServiceStub = DatasetServiceGrpc.newBlockingStub(client2Channel);
+    repositoryServiceStub = VersioningServiceGrpc.newBlockingStub(channel);
+    client2RepositoryServiceStub = VersioningServiceGrpc.newBlockingStub(client2Channel);
+  }
+
+  @AfterClass
+  public static void removeServerAndService() {
+    App.initiateShutdown(0);
+
+    // Remove all entities
+    // removeEntities();
+    // Delete entities by cron job
+    deleteEntitiesCron.run();
+
+    // shutdown test server
+    serverBuilder.build().shutdownNow();
+  }
+
+  private static final Collection<Object[]> requestData =
+      Arrays.asList(
+          new Object[][] {
+            {
+              ResourceVisibility.ORG_CUSTOM,
+              CollaboratorPermissions.newBuilder().setCollaboratorType(CollaboratorType.READ_WRITE),
+              true,
+              true
+            },
+            {ResourceVisibility.ORG_DEFAULT, CollaboratorPermissions.newBuilder(), true, false},
+            {
+              ResourceVisibility.ORG_CUSTOM,
+              CollaboratorPermissions.newBuilder().setCollaboratorType(CollaboratorType.READ_ONLY),
+              true,
+              false
+            },
+            {ResourceVisibility.PRIVATE, CollaboratorPermissions.newBuilder(), false, false}
+          });
+
+  @Test
+  public void createProjectWithGlobalSharingOrganization() {
+    LOGGER.info("Global organization Project test start................................");
+
+    if (app.getAuthServerHost() == null || app.getAuthServerPort() == null) {
+      Assert.assertTrue(true);
+      return;
+    }
+
+    String orgName = "Org-test-verta";
+    SetOrganization setOrganization =
+        SetOrganization.newBuilder()
+            .setOrganization(
+                Organization.newBuilder()
+                    .setName(orgName)
+                    .setDescription("This is the verta test organization")
+                    .build())
+            .build();
+    SetOrganization.Response orgResponse =
+        organizationServiceBlockingStub.setOrganization(setOrganization);
+    Organization organization = orgResponse.getOrganization();
+    assertEquals(
+        "Organization name not matched with expected organization name",
+        orgName,
+        organization.getName());
+
+    String orgRoleName = "O_" + organization.getId() + "_GLOBAL_SHARING";
+    GetRoleByName getRoleByName =
+        GetRoleByName.newBuilder()
+            .setName(orgRoleName)
+            .setScope(RoleScope.newBuilder().setOrgId(organization.getId()).build())
+            .build();
+    GetRoleByName.Response getRoleByNameResponse =
+        roleServiceBlockingStub.getRoleByName(getRoleByName);
+    assertEquals(
+        "Expected role name not found in DB",
+        orgRoleName,
+        getRoleByNameResponse.getRole().getName());
```

Comment:
Why are we testing role name from modeldb tests?

Reply:
I moved the test with this code from ProjectTest. I guess because it is test that checks integration between MDB and UAC

Reply:
Ok. I think mdb shouldn't worry about roles anymore, but we can revisit this later.